modules/social_features/social_landing_page/src/Plugin/Validation/Constraint/SocialAnLinkAccessValidator.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Drupal\social_landing_page\Plugin\Validation\Constraint;
+
+use Drupal\link\Plugin\Validation\Constraint\LinkAccessConstraintValidator;
+use Drupal\user\Entity\User;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * Validates the SocialAnLinkAccess constraint.
+ */
+class SocialAnLinkAccessValidator extends LinkAccessConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    if ($value->getFieldDefinition()->getName() !== 'field_button_link_an') {
```

Comment:
@dmytriikaiun could you add an explanation comment here?

Reply:
@ribel Added comments to the code.